actionners/kubernetes/drain/drain.go

```diff
@@ -189,6 +214,54 @@ func Action(action *rules.Action, event *events.Event) (utils.LogLine, *model.Da
 	}, nil, fmt.Errorf("the node '%v' has not been fully drained: %v eviction errors, %v other errors", nodeName, evictionErrorsCount, otherErrorsCount)
 }
 
+func verifyEvictionHasFinished(c *kubernetes.Client, period int, nodeName string, config Config) error {
+	tickerTiming := period / AmountOfTickers
+
+	timeout := time.After(time.Duration(period) * time.Second)
+	ticker := time.NewTicker(time.Duration(tickerTiming) * time.Second)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-timeout:
+			return errors.New("timeout reached before eviction finished")
+		case <-ticker.C:
+
+			var nonDaemonSetPods []string
+			excludedNamespaces := make(map[string]bool)
+			for _, namespace := range config.WaitPeriodExcludedNamespaces {
+				excludedNamespaces[namespace] = true
+			}
+			pods, err := c.CoreV1().Pods("").List(context.Background(), metav1.ListOptions{
+				FieldSelector: fmt.Sprintf("spec.nodeName=%s", nodeName),
+			})
+			if err != nil {
+				return err
+			}
+
+			for _, pod := range pods.Items {
+				isDaemonSet := false
```

Comment:
it also requires the statefulsets

Reply:
the logic here is to release the "holder" to the next action whenever there are no pods running on the node or all the pods running are daemonsets that we can ignore; why would we have StatefulSets in the logic too? We want to wait StatefulSets to be evicted before going to the next action.